### PR TITLE
Add support to relative paths as valid paths for interpreters config

### PR DESCRIPTION
**Pull request by fpliger**
Currently, we only allow full urls as versions of a specific interpreter, so something like:

``` html
<script type="micropython" version="./micropython.mjs">
```

is not allowed. This simple PR adds strings starting with `./` as valid options as well so that we can allow interpreters being server from `self`

### CODE REVIEW

**Comment by WebReflection** [comment]
I think `../` should allowed too?

File: esm/interpreters.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export const interpreter = new Proxy(new Map(), {
         if (!map.has(id)) {
             const [type, ...rest] = id.split('@');
             const interpreter = registry.get(type);
-            const url = /^https?:\/\//i.test(rest)
+            const url = /^(\.\/|https?:\/\/)/i.test(rest)
```
Comment (WebReflection) [inline esm/interpreters.js:24]: `const url = /^(?:\.\.?\/|https?:\/\/)/i.test(rest)` would be better here

**Review by WebReflection — CHANGES_REQUESTED** [review]
One tiny yet important detail + *maybe* a test would be cool (or a smoke test) ... but then also maybe I should add an interpreter mock to simplify the test or this repo life?

File: esm/interpreters.js (hunk 1 above)
Comment (fpliger) [reply esm/interpreters.js:24]: Yup, good point

**Comment by fpliger** [comment]
> One tiny yet important detail + _maybe_ a test would be cool (or a smoke test) ... but then also maybe I should add an interpreter mock to simplify the test or this repo life?

Yes. It was late when I put this PR yesterday and couldn't really think of a good and quick way to test this, so was planning on quickly chatting with you on it this morning.

If it works for you too, I was thinking that of adding a local build of `mpy` and loading that is a less than nothing smoke test, to merge this PR. Ultimately, I'd love to bring some test automation sometime soon.

If this works for you too, I'll go ahead with it

**Review by WebReflection — APPROVED** [review]
Perfect!

**Pull request merged**